File: src/core/interfaces/types/wizard.ts
import { ValidationError } from './errors';

export interface WizardState {
  id: string;
  type: string;
  currentStep: string;
  steps: WizardStep[];
  data: Record<string, any>;
  isComplete: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface WizardStep {
  id: string;
  title: string;
  description?: string;
  fields: WizardField[];
  validations?: Record<string, WizardValidationRule[]>;
  nextStep?: string;
  previousStep?: string;
  isOptional?: boolean;
  properties?: Record<string, any>;
}

export interface WizardValidation {
  isValid: boolean;
  errors: ValidationError[];
  warnings?: ValidationError[];
  suggestions?: string[];
}

export enum WizardStepType {
  FORM = 'FORM',
  SELECTION = 'SELECTION',
  CONFIRMATION = 'CONFIRMATION',
  SUMMARY = 'SUMMARY',
}

export interface WizardStepConfig {
  type: WizardStepType;
  validationRules?: Record<string, any>;
  requiredFields?: string[];
  customValidation?: (data: any) => Promise<WizardValidation>;
  onComplete?: (data: any) => Promise<void>;
}

export interface WizardValidationRule {
  type: 'required' | 'minLength' | 'maxLength' | 'pattern' | 'custom';
  value?: any;
  message: string;
}

export interface WizardField {
  id: string;
  type: 'text' | 'number' | 'select' | 'multiselect' | 'checkbox' | 'radio' | 'custom';
  label: string;
  placeholder?: string;
  defaultValue?: any;
  options?: Array<{ value: any; label: string }>;
  validations?: WizardValidationRule[];
  properties?: Record<string, any>;
}

export interface WizardCreateDTO {
  type: string;
  steps: WizardStep[];
  initialData?: Record<string, any>;
}

export type WizardUpdateDTO = Partial<WizardCreateDTO>;

File: backend/app/04_very_high_complexity/src/core/interfaces/types/wizard.py
from typing import Any, Dict, List, Union
from enum import Enum



class WizardState:
    id: str
    type: str
    currentStep: str
    steps: List[WizardStep]
    data: Dict[str, Any>
    isComplete: bool
    createdAt: str
    updatedAt: str
class WizardStep:
    id: str
    title: str
    description?: str
    fields: List[WizardField]
    validations?: Dict[str, WizardValidationRule[]>
    nextStep?: str
    previousStep?: str
    isOptional?: bool
    properties?: Dict[str, Any>
class WizardValidation:
    isValid: bool
    errors: List[ValidationError]
    warnings?: List[ValidationError]
    suggestions?: List[str]
class WizardStepType(Enum):
    FORM = 'FORM'
    SELECTION = 'SELECTION'
    CONFIRMATION = 'CONFIRMATION'
    SUMMARY = 'SUMMARY'
class WizardStepConfig:
    type: \'WizardStepType\'
    validationRules?: Dict[str, Any>
    requiredFields?: List[str]
    customValidation?: (data: Any) => Awaitable[WizardValidation>
    onComplete?: (data: Any) => Awaitable[None>
class WizardValidationRule:
    type: Union['required', 'minLength', 'maxLength', 'pattern', 'custom']
    value?: Any
    message: str
class WizardField:
    id: str
    type: Union['text', 'float', 'select', 'multiselect', 'checkbox', 'radio', 'custom']
    label: str
    placeholder?: str
    defaultValue?: Any
    options?: List[{ value: Any
    label: str>
  validations?: WizardValidationRule[]
  properties?: Record<string, any>
}
class WizardCreateDTO:
    type: str
    steps: List[WizardStep]
    initialData?: Dict[str, Any>
WizardUpdateDTO = Partial[WizardCreateDTO]